sources/packages/backend/apps/queue-consumers/src/services/cas-supplier/cas-supplier.service.ts

```diff
@@ -38,10 +50,10 @@ export class CASSupplierIntegrationService {
     const summary = new ProcessSummary();
     parentProcessSummary.children(summary);
     try {
-      summary.info("Logging on CAS...");
+      summary.info("Logging on CAS.");
       const auth = await this.casService.logon();
       summary.info("Logon successful.");
-      suppliersUpdated = await this.requestCASAndUpdateSuppliers(
+      suppliersUpdated = await this.processSuppliers(
```

Comment:
The silent failing is also happening inside `this.processSuppliers`.

Does this method need to fail silently w/o throwing error?

Reply:
The below AC is not implemented yet.
"If some error is present throw in the scheduler to allow it to be retried."
The idea is to inspect the summary (as it is already done in `getSuccessMessageWithAttentionCheck`) and throw an error if any error log message is present. Throwing an exception from here or from there would result in the same.

Reply:
I think the idea is that the failure will be caught and then logged for info purpose as `summary.error` and then in the next run of the scheduler, the failed cas suppliers will be attempted again.

Reply:
If some error is logged to the summary we will throw to allow the scheduler to retry immediately.
When we have the permanent errors implemented, those should be probably logged as warnings because they will be part of the "business valid scenarios".